src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/entity/Licence.kt

```diff
@@ -0,0 +1,102 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity
+
+import org.hibernate.annotations.Fetch
+import org.hibernate.annotations.FetchMode
+import org.hibernate.annotations.LazyCollection
+import org.hibernate.annotations.LazyCollectionOption
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus.IN_PROGRESS
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceType
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceType.AP
+import java.time.LocalDate
+import java.time.LocalDateTime
+import javax.persistence.Entity
+import javax.persistence.EnumType.STRING
+import javax.persistence.Enumerated
+import javax.persistence.GeneratedValue
+import javax.persistence.GenerationType
+import javax.persistence.Id
+import javax.persistence.JoinColumn
+import javax.persistence.OneToMany
+import javax.persistence.OrderBy
+import javax.persistence.Table
+import javax.validation.constraints.NotNull
+
+@Entity
+@Table(name = "licence")
+data class Licence(
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  @NotNull
+  val id: Long = -1,
+
+  @NotNull
+  @Enumerated(STRING)
+  val typeCode: LicenceType = AP,
+
+  val version: String? = null,
```

Comment:
Since we need to make comparisons on version number, should we store the version numbers as a integer so that we can do math on it? e.g. `if licence.version > currentVersion`. That's assuming we are using digits as version codes

Reply:
Might need major/minor versions, like `1.4`, so maybe a float type?

Reply:
Convention seems to be to keep version numbers as strings, since floating point numbers have poor precision. Lets leave it as is